src/table/header-cell/index.tsx

```diff
@@ -70,6 +72,7 @@ export function TableHeaderCell({
         [styles['header-cell-disabled']]: sortingDisabled,
         [styles['header-cell-ascending']]: sortingStatus === 'ascending',
         [styles['header-cell-descending']]: sortingStatus === 'descending',
+        [styles['header-cell-hidden']]: hidden,
```

Comment:
Does introducing a hidden state change the experience for screen-reader/assistive-technology?

Reply:
The "hidden" state already exists in the code, this change just modifies the border colour of those hidden cells.
When the sticky table header is enabled, there are actually _two_ headers: [one that scrolls with the table and keeps the layout correct](https://github.com/cloudscape-design/components/blob/main/src/table/internal.tsx#L231-L233), and [a duplicated header that stays stuck to the screen](https://github.com/cloudscape-design/components/blob/main/src/table/internal.tsx#L182-L193). From a visual standpoint, only one header is visible. The duplicated header [is also hidden from assistive technologies](https://github.com/cloudscape-design/components/blob/main/src/table/sticky-header.tsx#L53).
There is a short moment where technically both headers are visible, which happens when you scroll just a bit so that the font size of the page heading changes - there is an (intentionally) empty space between the now-stuck column headers and the first row. This empty space is made up of these "hidden" cells, and they should not have any border (in contrast to the non-hidden cells of the now-stuck header).